libsharedringbuffer/include/shared_ringbuffer/shared_ringbuffer.h

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2022, UNSW (ABN 57 195 873 179)
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#pragma once 
+
+#include <stdint.h>
+#include <stddef.h>
+#include <sel4utils/sel4_zf_logif.h>
+#include <utils/util.h>
+#include <shared_ringbuffer/gen_config.h>
+
+/* Function pointer to be used to 'notify' components on either end of the shared memory */
+typedef void (*notify_fn)(void);
+
+/* Buffer descriptor */
+typedef struct buff_desc {
+    uintptr_t encoded_addr; /* encoded dma addresses */
```

Comment:
Actually, this is agnostic of what this memory is. IN the end, this could also be a `void*` in the end.